h2o-algos/src/test/java/hex/tree/dt/DTTest.java

```diff
@@ -0,0 +1,390 @@
+            dt.trainModel().get();
+            fail("should have thrown validation error");
+        } catch (H2OModelBuilderIllegalArgumentException e) {
+            assertTrue(e.getMessage().contains("Only categorical response is supported"));
+            assertTrue(e.getMessage().contains("Only binary response is supported"));
+        } finally {
+            Scope.exit();
+        }
+    }
+
+    private void writePredictionsToFile(String path, Vec predictions, String prediction_column) {
+        File csvOutputFile = new File(path);
+        try (PrintWriter pw = new PrintWriter(csvOutputFile)) {
+            pw.println(prediction_column);
+            IntStream.range(0, (int) predictions.length()).mapToDouble(predictions::at)
+                    .mapToLong(Math::round).forEach(pw::println);
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+        assertTrue(csvOutputFile.exists());
+    }
+
+    @Test
+    public void testProstateSmallData() {
+
+        Scope.enter();
+        Frame train = Scope.track(parseTestFile("smalldata/testng/prostate_train.csv"));
+        Frame test = Scope.track(parseTestFile("smalldata/testng/prostate_test.csv"));
+        train.replace(0, train.vec(0).toCategoricalVec()).remove();
+        test.replace(0, test.vec(0).toCategoricalVec()).remove();
+        DKV.put(train);
+        DKV.put(test);
+
+        DTModel.DTParameters p =
+                new DTModel.DTParameters();
+        p._train = train._key;
+        p._seed = 0xDECAF;
+        p._max_depth = 12;
+        p._response_column = "CAPSULE";
+
+        DRFModel.DRFParameters p1 =
+                new DRFModel.DRFParameters();
+        p1._ntrees = 1;
+        p1._max_depth = 12;
+        p1._response_column = "CAPSULE";
+        p1._train = train._key;
+        p1._seed = 0xDECAF;
+
+        testDataset(train, test, p, p1, "prostate");
+        
+    }
+
+    @Test
+    public void testAirlinesSmallData() {
+        Scope.enter();
+        Frame train = Scope.track(parseTestFile("smalldata/testng/airlines_train_preprocessed.csv"));
+        Frame test = Scope.track(parseTestFile("smalldata/testng/airlines_test_preprocessed.csv"));
+        train.replace(0, train.vec(0).toCategoricalVec()).remove();
+        test.replace(0, test.vec(0).toCategoricalVec()).remove();
+        DKV.put(train);
+        DKV.put(test);
+
+        DTModel.DTParameters p =
+                new DTModel.DTParameters();
+        p._train = train._key;
+        p._seed = 0xDECAF;
+        p._max_depth = 12;
+        p._response_column = "IsDepDelayed";
+
+        DRFModel.DRFParameters p1 =
+                new DRFModel.DRFParameters();
+        p1._ntrees = 1;
+        p1._max_depth = 12;
+        p1._response_column = "IsDepDelayed";
+        p1._train = train._key;
+        p1._seed = 0xDECAF;
+
+        testDataset(train, test, p, p1, "airlines");
+        
+    }
+    
+
+    @Test
+    @Ignore
```

Comment:
Please use message why is it ignore - is it not working or is it take too long to complete and we have another _large python test for this pupose?